.github/workflows/test-deploy.yml

```diff
@@ -176,11 +185,37 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: '8.2'
+          tools: composer:v2
+
+      - name: Get Composer cache directory
+        id: composer-cache
+        run: echo "dir=$(composer config cache-files-dir)" >> $GITHUB_OUTPUT
+
+      - name: Cache Composer packages
+        uses: actions/cache@v4
+        with:
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: ${{ runner.os }}-composer-${{ hashFiles('composer.lock') }}
+          restore-keys: ${{ runner.os }}-composer-
+
       - name: Setup Node.js
         uses: actions/setup-node@v4
         with:
           node-version-file: '.nvmrc'
-          cache: 'npm'
+
+      - name: Cache npm dependencies
+        uses: actions/cache@v4
+        with:
+          path: |
+            node_modules
+            themes/*/node_modules
+          key: ${{ runner.os }}-node-${{ hashFiles('package-lock.json', '**/package.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
 
       - name: Install Node dependencies
         run: npm install --ignore-scripts
```

Comment:
If the node_modules were restored from the cache, there's no need to npm install again, as this will take unneeded time. 

```
- name: Install Node dependencies
        # We are checking for the root node_modules folder to confirm the cache was found.
        # The npm clean install deletes the node_modules folder and installs a fresh copy,
        # therefore we're skipping the command when the cache was hit.
        # Important: keep the list of node_modules folders in sync with the cache path above.
        run: |
          if [ -d "node_modules" ]; then
            echo "Node modules cache hit"
          else
            echo "Node modules cache miss"
            npm ci --cache .npm --prefer-offline --no-audit
          fi
```

Reply:
Are you saying your suggestion is a next step of the cache configuration, or required for the cache to work? Because right now, as-is, the time was drastically reduced from ~2m to ~15s on the `Install Node dependencies` step.

https://github.com/xwp/vip-site-template/actions/runs/17768434335/job/50497940366
<img width="1653" height="330" alt="image" src="https://github.com/user-attachments/assets/ed1e5b86-f7ff-46ea-a1c7-90fd2f4a80ae" />

https://github.com/xwp/vip-site-template/actions/runs/17768434335/job/50498519589
<img width="1657" height="334" alt="image" src="https://github.com/user-attachments/assets/d685451a-5d7e-4b12-94ae-3583e5afef6e" />

Reply:
That will cut those ±15s to 0s without any loss of functionality (or from ±40s if you have a lot more packages to 0 just as well, moving from O(n) complexity task down to O(1) ). It's the next step, and yes, it builds on top of the caching you got working of course.